### DISCUSSION TITLE
Tree structure realization for entities

### DISCUSSION THREAD

**Discussion by pincman** [mikro-orm/mikro-orm#2685] [Ideas]
**Is your feature request related to a problem? Please describe.**
tree entity support

**Describe the solution you'd like**
Self-association through parent and children for entity

**Describe alternatives you've considered**

**Additional context**
like [typeorm tree entities](https://typeorm.io/#/tree-entities)

**Comment by ml1nk** [comment]
Adjacency lists work like in typeorm and for more efficient queries we combine it with ltree (Materialized Path), which works nicely on mikro-orm. Closure tables and Nested sets should also be possible.

What functionality are you missing?

**Reply by mareksuscak** [reply]
@ml1nk could you share an example of what it takes to set up ltree support in MikroORM? Did you have to write [a custom type](https://mikro-orm.io/docs/custom-types) converter? Do you use raw queries to query data that uses the `ltree`?

**Comment by B4nan** [comment]
I also don't know what is missing, I am not planning to add any special decorators or what not like in typeorm, as I don't seem to understand why we would need such thing. You already can form a tree with your entities, relations can be as well self referencing or even cause cycles. 

Will move this into a discussion, as feature requests should have a clear description, and this is just a link to another tool's docs without any context.

**Reply by FranciscoMessina** [reply]
Maybe there could be an example in the docs added? To make it clearer that is is possible to achieve the behaviour.

**Comment by superiums** [comment]
the data structor of closure table is good for simplify the read and wirte operation of tree.

the main purpose of `@ClosureTable` , take this road more brieaf.

It's very similar as `@PivotEntry` in mikroorm.
```
@Entity()
export class Categeory {

  @OneToMany({entity:()=>Categeory})
  parent: Categoery

  @OneToMany({ entity: () => Categeory, pivotEntity: () => CatClosure })
  children = new Collection<Categoery>(this);

}
```

```
@Entity()
export class CatClosure {

  @ManyToOne({ primary: true })
  ancestor: Categeory;

  @ManyToOne({ primary: true })
  desendent: Categeory;

  @Property({ default: 1 })
  distance!: number;

}
```

through this, we may simulate the closure table of typeorm.
but there's more things to do to simplify the searching, the write to this data structure.

**Reply by B4nan** [reply]
See https://github.com/axotion/mikroorm-nested-set and https://gist.github.com/alexy4744/50edc40d2ed6eb706f2fcf3027dbe806
